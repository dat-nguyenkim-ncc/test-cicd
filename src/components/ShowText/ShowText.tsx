import React, { useState } from 'react'
import { Link } from 'theme-ui'
import { Paragraph } from '../primitives'

type ShowTextProps = {
  value: string
  max: number
}
const ShowText = ({ value, max }: ShowTextProps) => {
  const [showFull, setShowFull] = useState(false)

  return (
    <>
      <Paragraph sx={{ mb: 2, wordWrap: 'break-word' }}>
        {showFull ? value : `${value.slice(0, max)}${value.length > max ? '...' : ''}`}
      </Paragraph>
      {value.length >= max && (
        <Link
          href=""
          onClick={e => {
            e.preventDefault()
            setShowFull(!showFull)
          }}
        >
          {showFull ? 'Show less' : 'Show more'}
        </Link>
      )}
    </>
  )
}

export default ShowText
