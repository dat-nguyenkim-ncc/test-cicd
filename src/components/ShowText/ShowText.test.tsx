import * as React from 'react'
import { shallow } from 'enzyme'
import ShowText from './ShowText'

describe('ShowText', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(<ShowText value="Lorem ipsum dolot si amet" max={10} />)
    expect(wrapper).toMatchSnapshot()
  })
})
