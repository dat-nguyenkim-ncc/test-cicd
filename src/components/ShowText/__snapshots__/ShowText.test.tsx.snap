// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowText matches snapshot 1`] = `ShallowWrapper {}`;
